util/context/context.go

```diff
@@ -0,0 +1,35 @@
+// Package context contains common functions for storing and retrieving information from
+// standard go context
+package context
+
+import (
+	"context"
+
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type objectIdentifier string
+
+const (
+	name      objectIdentifier = `object_name`
+	namespace objectIdentifier = `object_namespace`
+)
+
+func InjectObjectMeta(ctx context.Context, meta *meta.ObjectMeta) context.Context {
+	ctx = context.WithValue(ctx, name, meta.Name)
+	return context.WithValue(ctx, namespace, meta.Namespace)
+}
+
+func ObjectName(ctx context.Context) string {
+	if n, ok := ctx.Value(name).(string); ok {
+		return n
+	}
+	return ""
+}
```

Comment:
I'd personally prefer returning a boolean as well.
Exactly like how maps work.

I know this is guaranteed to have this key though. 

But we do check the ok in the cast, continuing on with this lack of assuming things, we should also return an `ok` bool.

Reply:
When someone needs it we can add that, right now it wouldn't be used, so not adding in this commit.